lib/parallel.rb

```diff
@@ -469,7 +469,9 @@ def process_incoming_jobs(read, write, job_factory, options, &block)
         item, index = job_factory.unpack(data)
         result = begin
           call_with_index(item, index, options, &block)
-        rescue
+        rescue SystemExit
```

Comment:
```suggestion
        rescue SystemExit, SignalException
```
add SignalException so we do not capture the user hitting Ctrl+c ?